airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/state_converters/abstract_stream_state_converter.py

```diff
@@ -18,29 +18,30 @@ class AbstractStreamStateConverter(ABC):
     START_KEY = "start"
     END_KEY = "end"
 
-    def get_concurrent_stream_state(
-        self, cursor_field: Optional["CursorField"], state: MutableMapping[str, Any]
-    ) -> Optional[MutableMapping[str, Any]]:
-        if not cursor_field:
-            return None
-        if self.is_state_message_compatible(state):
-            return self.deserialize(state)
-        return self.convert_from_sequential_state(cursor_field, state)
-
     @abstractmethod
     def deserialize(self, state: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
         """
         Perform any transformations needed for compatibility with the converter.
         """
         ...
 
+    @abstractmethod
+    def get_sync_start(self, cursor_field: "CursorField", stream_state: MutableMapping[str, Any], start: Optional[Any]) -> Any:
```

Comment:
nit: Is there a case where we would call `get_sync_start` without calling `convert_from_sequential_state`? Else, I think I would merge the two to expose as few things as possible

Reply:
I like this idea! Updated to make this change.